rancher2/resource_rancher2_app.go

```diff
@@ -178,13 +190,26 @@ func resourceRancher2AppUpdate(d *schema.ResourceData, meta interface{}) error {
 	if d.Get("wait").(bool) {
 		stateConf := &resource.StateChangeConf{
 			Pending:    []string{},
-			Target:     []string{"active"},
+			Target:     []string{"installing"},
 			Refresh:    appStateRefreshFunc(client, id),
 			Timeout:    d.Timeout(schema.TimeoutUpdate),
 			Delay:      1 * time.Second,
 			MinTimeout: 3 * time.Second,
 		}
 		_, waitErr := stateConf.WaitForState()
+		if waitErr != nil {
+			return fmt.Errorf(
```

Comment:
One possible issue here is if the app actually finishes installing before the first refresh. I'd suggest setting a shorter timeout on this first `StateChangeConf` and log/ignore the `WaitForState` error here. That is unless there's a greater possibility that Rancher is responding "active" before install begins for a potentially long or infinite time.

Reply:
App should be on `installing` or `deploying` state before be created or updated. There is no delay on first state check, and refresh interval is adjusted to 2s, helm installation should last less than 2s to be avoided.

Reply:
Ok, sounds good. One more thing though:

In the block of code above this, you have a conditional call to the Rancher API, so it is possible that the resource is updated without calling the Rancher API. In that case this `WaitForState` will fail because the app was not modified.

Reply:
That's true, thanks. Updated

Reply:
Finally i've changed the scope. App first is waiting until it's not transitioning, and then checking for active state. This should be more accurate.